File: java/1TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; ++i) {
            if (map.containsKey(target - nums[i])) {
                int index = map.get(target - nums[i]);
                return new int[] {index, i};
            }
            
            // add entry nums[i] -> i to the map
            map.put(nums[i], i);
        }
        
        return new int[] {};
    }
}

File: java/1TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for i, num in enumerate(nums):
            d[num] = i
        for i, num in enumerate(nums):
            ix = d.get(target - num)
            if ix and ix != i:
                resp = [i, d[target - num]]
                return resp
